DQM/CTPPS/plugins/CTPPSDiamondDQMSource.cc

```diff
@@ -686,9 +686,18 @@ std::shared_ptr<dds::Cache> CTPPSDiamondDQMSource::globalBeginLuminosityBlock(co
   auto d = std::make_shared<dds::Cache>();
   d->hitDistribution2dMap.reserve(potPlots_.size());
   if (!perLSsaving_ && plotOnline_) {
-    for (auto& plot : potPlots_)
-      d->hitDistribution2dMap[plot.first] =
-          std::unique_ptr<TH2F>(static_cast<TH2F*>(plot.second.hitDistribution2d_lumisection->getTH2F()->Clone()));
+    for (auto& plot : potPlots_) {
+      d->hitDistribution2dMap[plot.first] = std::make_unique<TH2F>(
+          "hits in planes lumisection",
+          (std::string(plot.second.hitDistribution2d_lumisection->getTH2F()->GetTitle()) + ";plane number;x (mm)")
+              .c_str(),
```

Comment:
I believe this leads to use-after-free. The return value of `c_str()`, i.e. the pointer,  is kept alive until the `make_unique` returns, but the temporary string is not.

Reply:
I am afraid I do not follow, of course this array is kept alive, c_str() is just returning pointer to char* which is a buffer of a string with trailing null, you can imagine there's no c_str() invocation, then you are expecting this argument to be deallocated before it is passed to a function? @makortel  can you please clarify, maybe I am not understanding this correctly.

Reply:
Sorry for the noise, I was wrong. The temporary string is indeed kept alive until the `make_unique` returns.